### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by Quiet-Klirik**
**Manufacturer List:**

image

**Car List:**

image

**Car Detail:**

image

**Driver List:**

image

**Driver Detail:**

image

### CODE REVIEW

File: taxi/views.py [hunk 1]
```diff
@@ -17,3 +18,46 @@ def index(request):
...
+    model = Driver
+    paginate_by = 5
+    context_object_name = "driver_list"
+    template_name = "taxi/driver_list.html"
+
+
+class DriverDetailView(generic.DetailView):
+    model = Driver
+    context_object_name = "driver"
+    template_name = "taxi/driver_detail.html"
```
Comment (pavlejviki) [inline taxi/views.py:58]: Optimize queryset: don't make a query for a manufacturer for each car (N+1 problem).

**Review by pavlejviki — CHANGES_REQUESTED** [review]

File: taxi/views.py [hunk 2]
```diff
@@ -17,3 +18,46 @@ def index(request):
...
+
+    def get_queryset(self):
+        return self.model.objects.prefetch_related(
+            "cars", "cars__manufacturer"
+        ).filter(id=self.kwargs.get("pk"))
```
Comment (pavlejviki) [inline taxi/views.py:63]: you can perform the optimisation in the class attribute 
 queryset = Driver.objects.prefetch_related("cars__manufacturer")

File: taxi/views.py [hunk 3]
```diff
@@ -17,3 +18,46 @@ def index(request):
     }
 
     return render(request, "taxi/index.html", context=context)
+
+
+class ManufacturerListView(generic.ListView):
+    # queryset = Manufacturer.objects.order_by("name")
+    model = Manufacturer
+    ordering = ["name"]
```
Comment (pavlejviki) [inline taxi/views.py:26]: replace ordering by your commented attribute

```suggestion
    queryset = Manufacturer.objects.order_by("name")
```

**Review by pavlejviki — CHANGES_REQUESTED** [review]

File: taxi/urls.py [hunk 4]
```diff
@@ -1,9 +1,35 @@
 from django.urls import path
 
-from .views import index
+from .views import index, ManufacturerListView, CarListView, CarDetailView, \
```
Comment (Polyakiv-Andrey) [inline taxi/urls.py:3]: use brackets instead of slash

**Review by Polyakiv-Andrey — CHANGES_REQUESTED** [review]

**Pull request closed without merging**